docs/features/nullable-reference-types.md

```diff
@@ -94,14 +94,25 @@ Invocation of methods annotated with the following attributes will also affect f
 - `[AssertsTrue]` (e.g. `Debug.Assert`) and `[AssertsFalse]`
 
 ## `default`
-`default(T)` is `T?` if `T` is a reference type.
-_Is `default(T)` also `T?` if `T` is an unconstrained type parameter?_
-_Is `default(T?)` an error?_
+If `T` is a reference type, `default(T)` is `T?`.
 ```c#
 string? s = default(string); // assigns ?, no warning
 string t = default; // assigns ?, warning
-T t = default; // assigns ?, warning
 ```
+If `T` is a value type, `default(T)` is `T` and any non-nullable fields in `T` are maybe null.
+If `T` is a value type, `new T()` is equivalent to `default(T)`.
```

Comment:
>`new T()` is equivalent to `default(T)` [](start = 24, length = 39)

Does this depend on whether we're using a default constructor, or a user-defined one?

Reply:
This only applies to the default parameter-less constructor that cannot be overridden for structs.

---
In reply to: [246863817](https://github.com/dotnet/roslyn/pull/32304#discussion_r246863817) [](ancestors = 246863817)

Reply:
Ah, I didn't know they couldn't be overridden. Thanks

---
In reply to: [246865325](https://github.com/dotnet/roslyn/pull/32304#discussion_r246865325) [](ancestors = 246865325,246863817)